pkg/crc/cluster/cluster.go

```diff
@@ -119,20 +120,82 @@ func UpdateClusterID(ocConfig oc.Config) error {
 	return nil
 }
 
-func AddProxyConfigToCluster(ocConfig oc.Config, proxy *network.ProxyConfig) error {
-	cmdArgs := []string{"patch", "proxy", "cluster", "-p",
-		fmt.Sprintf(`'{"spec":{"httpProxy":"%s", "httpsProxy":"%s", "noProxy":"%s"}}'`, proxy.HTTPProxy, proxy.HTTPSProxy, proxy.GetNoProxyString()),
-		"-n", "openshift-config", "--type", "merge"}
+func AddProxyConfigToCluster(sshRunner *ssh.Runner, ocConfig oc.Config, proxy *network.ProxyConfig) error {
+	type trustedCA struct {
+		Name string `json:"name"`
+	}
+
+	type proxyPatch struct {
```

Comment:
I would name this `proxyConfig`...

Reply:
there is already `ProxyConfig` struct in network so I don't want to have any misunderstanding it here, also struct is to create patch for proxy config so I OK with current naming.

Reply:
`specProxyConfig` maybe? I don't think this format would be only valid for `patch`? For example if we parse what `oc get` returns, we'd use the same struct?

Reply:
if `oc get proxy` then you can use same struct to extract the `spec` data or may be other resource which have same kind of data as part of spec, but I can go with `proxySpecConfig`